keps/sig-network/4969-cluster-domain-pod-api/README.md

```diff
@@ -0,0 +1,703 @@
+new status field will be set for new Pods launched from then on.
+
+When downgrading a cluster, the status field will not be set anymore for new
+Pods.
+
+### Version Skew Strategy
+
+Older kubelets (or kubelets with the feature disabled) will not set the attribute,
+and fail to create Pods that request the attribute via the downward API.
+
+Older apiservers will ignore attempts (by the kubelet) to set the attribute, and
+reject attempts to create Pods that request the attribute via the downward API.
+
+Applications that want to be compatible with both environments should have a
+provision for the field being unset (such as a hard-coded default value, and/or
+a manual override flag).
+
+Additionally, such applications should retrieve the value from the Kubernetes
+REST API, rather than using the downward API.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: PodClusterDomain
+  - Components depending on the feature gate:
+    - kubelet
+    - kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+No. Workloads that don't read the status field or request it via the downward
+API would be unaffected.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes.
+
+Workloads that expect the feature to be available (and don't have a fallback in
+place) would fail, but that has nothing to do with the rollout itself.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+It should become available again. Existing Pods should be unaffected.
+
+###### Are there any tests for feature enablement/disablement?
+
+Not yet. The logic should be pretty trivial, but it probably makes sense to at
+least test gate mismatches between apiserver and kubelet.
```

Comment:
Please describe what the tests will look like and if that's going to be integration (most likely, since you're trying to coordinate between kubelet and apiserver) or unit/e2e?